cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/StorageProviderNamespaceHandlerTest.java

```diff
@@ -29,16 +34,24 @@
  */
 public class StorageProviderNamespaceHandlerTest extends DatasetServiceTestBase {
 
+  private static final Gson GSON = new GsonBuilder().create();
+
   @Test
-  public void test() throws IOException {
-    Assert.assertEquals(200, createNamespace("myspace").getResponseCode());
-    // creation is idempotent, now that we delete the namespace directory if it exists
+  public void test() throws Exception {
+    // first add the namespace meta
+    namespaceAdmin.create(new NamespaceMeta.Builder().setName("myspace").build());
+    // creating the namesapce in underlying storage provider should work
     Assert.assertEquals(200, createNamespace("myspace").getResponseCode());
+    // Since the namespace directory already exist this creating the same should fail
+    Assert.assertEquals(500, createNamespace("myspace").getResponseCode());
+    // should be able to delete the namespace
     Assert.assertEquals(200, deleteNamespace("myspace").getResponseCode());
```

Comment:
Just realized I can add more unit test here specifically which test custom namespace mapping dir does not get create/deleted. Will do so.